packages/sdk/src/utils/spinner.ts

```diff
@@ -0,0 +1,75 @@
+import type { Options as OraOptions } from 'ora'
+import ora from 'ora'
+
+const defaultOraOptions: OraOptions = {
+  spinner: 'dots',
+  color: 'cyan',
+  indent: 0,
+  stream: process.stdout,
+}
+
+/**
+ * Methods available to a spinner instance that has already started.
+ */
+export interface SpinnerStarted {
+  success(text?: string): void
+  failure(text?: string): void
+}
+
+/**
+ * Closure that starts a spinner if `enableOutput` is true, and returns a `SpinnerStarted` instance.
+ * @param enableOutput Whether to enable or disable any output. Useful e.g. for "--print" flags in commands.
+ * @param oraOptions Additional options to pass to `ora` for customizing the spinner.
+ * @returns
+ */
+export function createSpinner(enableOutput = true, oraOptions: Partial<OraOptions> = {}) {
+  const actualOptions = { ...defaultOraOptions, ...oraOptions }
+
+  return (text: string): SpinnerStarted => {
+    if (!enableOutput) {
+      return {
+        success: () => {},
+        failure: () => {},
+      }
+    }
+
+    actualOptions.stream?.write('\n')
+
+    /**
+     * Ships the mocks with the actual implementation in order to have sane jest snapshots.
+     * This is a workaround because jest doesn't support mocking classes.
+     */
+    if (process.env.NODE_ENV === 'test' || process.env.CI) {
```

Comment:
🤔 So Jest here defaults to `NODE_ENV=test` if not already defined.

That also applied to users of the Prisma CLI which might have this env var set.

Maybe we could remove the `(spinner)` parts since it's a bit weird from user perspective to get that?

Reply:
The `(spinner)` text actually appears in the Jest snapshots only, the users will see a rotating thing in the CLI. I thought it could be useful so we can understand from snapshots where the user sees a spinner, vs where the user is left hanging on the terminal. Should I drop it?

Btw, you can test it directly in your terminal by running an introspection on any schema using `prisma@3.13.0-integration-add-spinner-to-introspection.1`.

Reply:
About the env var, I understand your concerns. Maybe I could drop the factory method altogether, expose a single spinner instance, and hope that we can mock it with jest (even though we've had issues some days ago with mocking functions exposed by other modules)

Reply:
Note from my DM: Looking at the docs it looks we could simplify since the lib already checks if the context is TTY / CI https://github.com/sindresorhus/ora/tree/v5.4.1#isenabled
```
isEnabled

Type: boolean

Force enable/disable the spinner. If not specified, the spinner will be enabled if the stream is being run inside a TTY context (not spawned or piped) and/or not in a CI environment.

Note that {isEnabled: false} doesn't mean it won't output anything. It just means it won't output the spinner, colors, and other ansi escape codes. It will still log text.
```